backends/portfolio/tasks.rake

```diff
@@ -152,7 +154,117 @@ def pf_adoc2html(adoc_file, target_pname)
   sh "chmod +x #{run_pname}"
 
   # Now run the actual command.
-  sh cmd
+  sh(cmd)
 
   $logger.info "Generated HTML in #{target_pname}"
 end
+
+# @param adoc_file [String] Full pathname of source adoc file
+# @param target_pname [String] Full name of tags file being generated
+# @param isa_manual_dirname [String] Full pathname of ISA manual root directory
+def pf_adoc2norm_tags(adoc_file, target_pname, isa_manual_dirname)
+  target_dirname = File.dirname(target_pname)
+
+  # Ensure target directory is present.
+  FileUtils.mkdir_p(target_dirname)
+
+  # The tags backend will put the tags file in the same directory as the input adoc file.
+  backend_tags_pname = adoc_file.sub(/\.adoc$/, "-norm-tags.json")
```

Comment:
since the string "-norm-tags.json" is used multiple times, it might help to use a variable, in case there is a need to change it later, so it can be changed once.